### ISSUE TITLE
Javascript not running

### ISSUE THREAD

**Issue by ameteshov** [wkhtmltopdf/wkhtmltopdf#4025]
Hi there. Javascript not running, and I have no idea why, could you please help? 
Html 
```html
<!doctype html>
<html>
<head>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/Chart.js/2.7.2/Chart.min.js"></script>
</head>
<body>
    <canvas id="myChart" width="400" height="400"></canvas>
    <script type="text/javascript">
        var ctx = document.getElementById("myChart").getContext('2d');

        var myChart = new Chart(ctx, {
            type: 'bar',
            data: {
                labels: ["Red", "Blue", "Yellow", "Green", "Purple", "Orange"],
                datasets: [{
                    label: '# of Votes',
                    data: [12, 19, 3, 5, 2, 3],
                    backgroundColor: [
                        'rgba(255, 99, 132, 0.2)',
                        'rgba(54, 162, 235, 0.2)',
                        'rgba(255, 206, 86, 0.2)',
                        'rgba(75, 192, 192, 0.2)',
                        'rgba(153, 102, 255, 0.2)',
                        'rgba(255, 159, 64, 0.2)'
                    ],
                    borderColor: [
                        'rgba(255,99,132,1)',
                        'rgba(54, 162, 235, 1)',
                        'rgba(255, 206, 86, 1)',
                        'rgba(75, 192, 192, 1)',
                        'rgba(153, 102, 255, 1)',
                        'rgba(255, 159, 64, 1)'
                    ],
                    borderWidth: 1
                }]
            },
            options: {
                scales: {
                    yAxes: [{
                        ticks: {
                            beginAtZero:true
                        }
                    }]
                }
            }
        });
    </script>
</body>
</html>
```
Command: 
`wkhtmltopdf --debug-javascript --javascript-delay 500 --enable-javascript --no-stop-slow-scripts graph.html test.pdf`
Version:
wkhtmltopdf 0.12.5 (with patched qt)
OS:
HighSierra 10.13.5

**Comment by vevisystems** [comment]
Javascript actually works, I think your problem is related with the setLineDash() funtion, already reported in issues #3938 or #3242.

Adding the following js code should fix the issue, but beware it disables setLineDash() so no dashed lines until the issue is actually fixed:

CanvasRenderingContext2D.prototype.setLineDash = function override$setLineDash(segments) { }

**Comment by ameteshov** [comment]
@vevisystems thank you for your answer, I'm sorry for delay. Unfortunately, your advice didn't help me. 
I think this issue should be closed because of related #3938 or #3242.

**Comment by nschappl** [comment]
I wasn’t able to get `wkhtmltopdf` to work for my use-cases either. I had problems with integrating ES6. Also, the javascript delay wasn’t useful when you had PDFs getting created with different amount of asynchronously loaded data — sometimes it only takes 1 second but other times it takes 10 seconds, and so it’s pretty inefficient to always wait 10 seconds for the PDF to generate.

After burning many hours trying to get `wkhtmltopdf` to work, I gave up and found a simple microservice for converting HTML (or a URL) to images/pdfs available on [Bantam.io](https://bantam.io). I got the PDF generation I needed there in a couple minutes.

```js
bantam
  .run('@images/html/url', {
    url: 'https://www.mapbox.com/mapbox-gl-js/style-spec/',
    waitUntil: 'networkidle0',
  })
  .then(pdfUrl => {
      // the PDF is temporarily hosted on AWS S3 via a secured URL at `pdfUrl`
      //    - you can download it, send the link to others, etc
      //    - if you need permanent storage, you can get the raw PDF data and
      //      pipe it somewhere by using the `request` node module like:
      //      `request.get(pdfUrl).pipe(res);`
    });
```

Here are the docs for it: https://bantam.io/functions/@images/html?link=docs&subLink=0

There are some awesome configs as well like the `networkidle0` above which waits for all network connections at the specified URL to stop for half a second before capturing the PDF (e.g. ensuring that data is loaded when the PDF is created without waiting an arbitrarily long period of time).

Maybe this can help some people that are still stuck.
